microservice/src/test/java/de/gsi/microservice/concepts/majordomo/legacy/CmwLightClient.java

```diff
@@ -0,0 +1,319 @@
+            System.out.println("received exception: " + data);
+        } else if (!Arrays.equals(data.getLast().getData(), new byte[] { MT_HEADER, MT_BODY, MT_BODY_DATA_CONTEXT })) {
+            System.out.println("expected reply message, but got: " + Arrays.toString(data.getLast().getData()));
+            return;
+        } else {
+            System.out.println("received reply: " + data);
+        }
+        for (final byte desc : descriptor) {
+            switch (desc) {
+            case MT_HEADER:
+                System.out.println("header: " + new String(data.pollFirst().getData()));
+                break;
+            case MT_BODY:
+                final byte[] bytes = data.pollFirst().getData();
+                final IoClassSerialiser classSerialiser = new IoClassSerialiser(FastByteBuffer.wrap(bytes), CmwLightSerialiser.class);
+                final SnoopAcquisition snoopAcq = classSerialiser.deserialiseObject(SnoopAcquisition.class);
+                System.out.println("body: " + snoopAcq);
+                break;
+            case MT_BODY_DATA_CONTEXT:
+                System.out.println("body data context: " + new String(data.pollFirst().getData()));
+                break;
+            case MT_BODY_REQUEST_CONTEXT:
+                System.out.println("body request context: " + new String(data.pollFirst().getData()));
+                break;
+            case MT_BODY_EXCEPTION:
+                System.out.println("body exception: " + new String(data.pollFirst().getData()));
+                break;
+            default:
+                System.out.println("invalid message type (" + desc + "): " + new String(data.pollFirst().getData()));
+            }
+        }
+    }
+
+    public void subscribeSnoopFromDigitizer() {
+        System.out.println("connect control socket");
+        controlChannel.connect("tcp://dal005.acc.gsi.de:13810");
+
+        connectClient();
+
+        subscribe("GSCD001", "SnoopTriggerEvents", "FAIR.SELECTOR.ALL");
+
+        while (true) {
+            receiveData();
+        }
+    }
+
+    public void getSnoopFromDigitizer() {
+        System.out.println("connect control socket");
+        controlChannel.connect("tcp://dal005.acc.gsi.de:13810");
+
+        connectClient();
+
+        sendHeartBeat();
+
+        get("GSCD001", "SnoopTriggerEvents", "FAIR.SELECTOR.S=4");
+
+        while (true) {
+            receiveData();
+        }
+    }
+
+    public void getLocal() {
+        System.out.println("connect control socket");
+        controlChannel.connect("tcp://SYSPC004:7777");
+
+        connectClient();
+
+        sendHeartBeat();
+
+        get("testdevice", "unknownProp", "FAIR.SELECTOR.ALL");
+
+        // return reply to data
+        while (true) {
+            final byte[] data = controlChannel.recv();
+            if (data != null && data.length != 0) {
+                if (Arrays.equals(data, new byte[] { MT_HEADER, MT_BODY_EXCEPTION })) { // header, exception
```

Comment:
Codacy found an issue: [These nested if statements could be combined](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=6307742)